From a77991c6a392c4865445506dc910ec3b7f7d49a8 Mon Sep 17 00:00:00 2001
From: Tux <write@imaginarycode.com>
Date: Thu, 5 Mar 2020 22:22:20 +0530
Subject: [PATCH] Resolve sendData() deadlocks


diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeServerInfo.java b/proxy/src/main/java/net/md_5/bungee/BungeeServerInfo.java
index 4bb0f68f..aed3b23d 100644
--- a/proxy/src/main/java/net/md_5/bungee/BungeeServerInfo.java
+++ b/proxy/src/main/java/net/md_5/bungee/BungeeServerInfo.java
@@ -110,19 +110,22 @@ public class BungeeServerInfo implements ServerInfo
         Preconditions.checkNotNull( channel, "channel" );
         Preconditions.checkNotNull( data, "data" );
 
-        synchronized ( packetQueue )
+        Server server;
+
+        synchronized ( players )
         {
-            Server server = ( players.isEmpty() ) ? null : players.iterator().next().getServer();
-            if ( server != null )
-            {
-                server.sendData( channel, data );
-                return true;
-            } else if ( queue )
-            {
-                packetQueue.add( new PluginMessage( channel, data, false ) );
-            }
-            return false;
+            server = ( players.isEmpty() ) ? null : players.iterator().next().getServer();
+        }
+        
+        if ( server != null )
+        {
+            server.sendData( channel, data );
+            return true;
+        } else if ( queue )
+        {
+            packetQueue.add( new PluginMessage( channel, data, false ) );
         }
+        return false;
     }
 
     private long lastPing;
-- 
2.24.0.windows.2

