From cf016fa853299348e9b828dd2d2c57991480b884 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=ABsFot?= <jesfot@gmail.com>
Date: Fri, 6 Mar 2020 14:26:32 +0530
Subject: [PATCH] Disable support for MC version < 1.14


diff --git a/protocol/src/main/java/net/md_5/bungee/protocol/ProtocolConstants.java b/protocol/src/main/java/net/md_5/bungee/protocol/ProtocolConstants.java
index 55911163..4ff963ee 100644
--- a/protocol/src/main/java/net/md_5/bungee/protocol/ProtocolConstants.java
+++ b/protocol/src/main/java/net/md_5/bungee/protocol/ProtocolConstants.java
@@ -29,30 +29,30 @@ public class ProtocolConstants
     public static final int MINECRAFT_1_15_1 = 575;
     public static final int MINECRAFT_1_15_2 = 578;
     public static final List<String> SUPPORTED_VERSIONS = Arrays.asList(
-            "1.8.x",
-            "1.9.x",
-            "1.10.x",
-            "1.11.x",
-            "1.12.x",
-            "1.13.x",
+            //"1.8.x",
+            //"1.9.x",
+            //"1.10.x",
+            //"1.11.x",
+            //"1.12.x",
+            //"1.13.x",
             "1.14.x",
             "1.15.x"
     );
     public static final List<Integer> SUPPORTED_VERSION_IDS = Arrays.asList(
-            ProtocolConstants.MINECRAFT_1_8,
-            ProtocolConstants.MINECRAFT_1_9,
-            ProtocolConstants.MINECRAFT_1_9_1,
-            ProtocolConstants.MINECRAFT_1_9_2,
-            ProtocolConstants.MINECRAFT_1_9_4,
-            ProtocolConstants.MINECRAFT_1_10,
-            ProtocolConstants.MINECRAFT_1_11,
-            ProtocolConstants.MINECRAFT_1_11_1,
-            ProtocolConstants.MINECRAFT_1_12,
-            ProtocolConstants.MINECRAFT_1_12_1,
-            ProtocolConstants.MINECRAFT_1_12_2,
-            ProtocolConstants.MINECRAFT_1_13,
-            ProtocolConstants.MINECRAFT_1_13_1,
-            ProtocolConstants.MINECRAFT_1_13_2,
+            //ProtocolConstants.MINECRAFT_1_8,
+            //ProtocolConstants.MINECRAFT_1_9,
+            //ProtocolConstants.MINECRAFT_1_9_1,
+            //ProtocolConstants.MINECRAFT_1_9_2,
+            //ProtocolConstants.MINECRAFT_1_9_4,
+            //ProtocolConstants.MINECRAFT_1_10,
+            //ProtocolConstants.MINECRAFT_1_11,
+            //ProtocolConstants.MINECRAFT_1_11_1,
+            //ProtocolConstants.MINECRAFT_1_12,
+            //ProtocolConstants.MINECRAFT_1_12_1,
+            //ProtocolConstants.MINECRAFT_1_12_2,
+            //ProtocolConstants.MINECRAFT_1_13,
+            //ProtocolConstants.MINECRAFT_1_13_1,
+            //ProtocolConstants.MINECRAFT_1_13_2,
             ProtocolConstants.MINECRAFT_1_14,
             ProtocolConstants.MINECRAFT_1_14_1,
             ProtocolConstants.MINECRAFT_1_14_2,
-- 
2.24.0.windows.2

